cogs/utils/paginator.py

```diff
@@ -0,0 +1,302 @@
+import asyncio
+import discord
+from discord.ext.commands import Paginator as CommandPaginator
+
+FIRST_EMOJI = "\u23EE"   # [:track_previous:]
+LEFT_EMOJI = "\u2B05"    # [:arrow_left:]
+RIGHT_EMOJI = "\u27A1"   # [:arrow_right:]
+LAST_EMOJI = "\u23ED"    # [:track_next:]
+PAGE_NUM_EMOJI = ":1234: "
+STOP_EMOJI = ":stop_button: "
+INFO_EMOJI = ":information_source: "
+
+class CannotPaginate(Exception):
+    pass
+
+
+class Pages:
+    """
+    Implements a paginator that queries the user for the
+    pagination interface.
+    Pages are 1-index based, not 0-index based.
+    If the user does not reply within 2 minutes then the pagination
+    interface exits automatically.
+    Parameters
+    ------------
+    ctx: Context
+        The context of the command.
+    entries: List[str]
+        A list of entries to paginate.
+    per_page: int
+        How many entries show up per page.
+    show_entry_count: bool
+        Whether to show an entry count in the footer.
+    Attributes
+    -----------
+    embed: discord.Embed
+        The embed object that is being used to send pagination info.
+        Feel free to modify this externally. Only the description,
+        footer fields, and colour are internally modified.
+    permissions: discord.Permissions
+        Our permissions for the channel.
+    """
+
+    def __init__(self, ctx, *, entries, per_page=12, show_entry_count=True):
+        self.bot = ctx.bot
+        self.entries = entries
+        self.message = ctx.message
+        self.channel = ctx.channel
+        self.author = ctx.author
+        self.per_page = per_page
+        pages, left_over = divmod(len(self.entries), self.per_page)
+        if left_over:
+            pages += 1
+        self.maximum_pages = pages
+        self.embed = discord.Embed(colour=discord.Colour.blurple())
+        self.paginating = len(entries) > per_page
+        self.show_entry_count = show_entry_count
+        self.reaction_emojis = [
+            (f'{FIRST_EMOJI}', self.first_page),
+            (f'{LEFT_EMOJI}', self.previous_page),
+            (f'{RIGHT_EMOJI}', self.next_page),
+            (f'{LAST_EMOJI}', self.last_page),
+            (f'{PAGE_NUM_EMOJI}', self.numbered_page),
+            (f'{STOP_EMOJI}', self.stop_pages),
+            (f'{INFO_EMOJI}', self.show_help),
```

Comment:
```suggestion
            (FIRST_EMOJI}, self.first_page),
            (LEFT_EMOJI}, self.previous_page),
            (RIGHT_EMOJI, self.next_page),
            (LAST_EMOJI, self.last_page),
            (PAGE_NUM_EMOJI, self.numbered_page),
            ({STOP_EMOJI, self.stop_pages),
            ({INFO_EMOJI, self.show_help),
```
No need to encapsulate these variables in f-strings they're strings already